osquery/database/tests/database_tests.cpp

```diff
@@ -23,31 +23,87 @@ namespace osquery {
 
 class DatabaseTests : public testing::Test {};
 
-TEST_F(DatabaseTests, test_set_value) {
-  auto s = setDatabaseValue(kLogs, "i", "{}");
+TEST_F(DatabaseTests, test_set_value_str) {
+  auto s = setDatabaseValue(kLogs, "str", "{}");
   EXPECT_TRUE(s.ok());
 }
 
-TEST_F(DatabaseTests, test_get_value) {
-  std::string expected = "{}";
-  setDatabaseValue(kLogs, "i", expected);
+TEST_F(DatabaseTests, test_set_value_int) {
+  auto s = setDatabaseValue(kLogs, "int", -1);
+  EXPECT_TRUE(s.ok());
+}
 
-  std::string value;
-  auto s = getDatabaseValue(kLogs, "i", value);
+TEST_F(DatabaseTests, test_set_value_mix1) {
+  auto s = setDatabaseValue(kLogs, "intstr", -1);
+  EXPECT_TRUE(s.ok());
 
+  s = setDatabaseValue(kLogs, "intstr", "{}");
   EXPECT_TRUE(s.ok());
-  EXPECT_EQ(value, expected);
+}
 
+TEST_F(DatabaseTests, test_set_value_mix2) {
+  auto s = setDatabaseValue(kLogs, "strint", "{}");
+  EXPECT_TRUE(s.ok());
+
+  s = setDatabaseValue(kLogs, "strint", -1);
+  EXPECT_TRUE(s.ok());
+}
+
+TEST_F(DatabaseTests, test_get_value_does_not_exist) {
   // Unknown keys return failed, but will return empty data.
-  value.clear();
-  s = getDatabaseValue(kLogs, "does_not_exist", value);
+  std::string value;
+  auto s = getDatabaseValue(kLogs, "does_not_exist", value);
   EXPECT_FALSE(s.ok());
   EXPECT_TRUE(value.empty());
 }
 
+TEST_F(DatabaseTests, test_get_value_str) {
+  std::string expected = "{}";
+  setDatabaseValue(kLogs, "str", expected);
+
+  std::string value;
+  auto s = getDatabaseValue(kLogs, "str", value);
+
+  EXPECT_TRUE(s.ok());
+  EXPECT_EQ(value, expected);
+}
+
+TEST_F(DatabaseTests, test_get_value_int) {
+  int expected = -1;
+  setDatabaseValue(kLogs, "int", expected);
+
+  int value;
```

Comment:
To leave value unassigned is no a good idea, according to standard the value is unspecified.
You could use `auto value = int{};` or `int value{};` or `int value = 0;` which is the same.